scapy/layers/can.py

```diff
@@ -106,7 +106,7 @@ def dispatch_hook(cls,
                 not orb(_pkt[5]) & 0xf8
             if fdf_set:
                 return CANFD
-            elif len(_pkt) > 16:
+            elif len(_pkt) > 4 and orb(_pkt[4]) > 8:
```

Comment:
Note: now that we've dropped 2.7, you actually don't need `orb` anymore (at all)